packages/features/bookings/lib/getBookingResponsesSchema.ts

```diff
@@ -20,18 +20,18 @@ const ensureValidPhoneNumber = (value: string) => {
   // Replace the space(s) in the beginning with + as it is supposed to be provided in the beginning only
   return value.replace(/^ +/, "+");
 };
-export const getBookingResponsesPartialSchema = ({ bookingFields, view }: CommonParams) => {
+export const getBookingResponsesPartialSchema = ({ bookingFields, view, prevResponse }: CommonParams) => {
   const schema = bookingResponses.unwrap().partial().and(catchAllSchema);
 
-  return preprocess({ schema, bookingFields, isPartialSchema: true, view });
+  return preprocess({ schema, bookingFields, isPartialSchema: true, view, prevResponse });
 };
 
 // Should be used when we know that not all fields responses are present
 // - Can happen when we are parsing the prefill query string
 // - Can happen when we are parsing a booking's responses (which was created before we added a new required field)
-export default function getBookingResponsesSchema({ bookingFields, view }: CommonParams) {
```

Comment:
I don't think it makes sense to pass `prevResponses` to a function that builds the schema.

If you would see this schema when built, is passed the responses but no responses are passed to it while building the schema.

Reply:
Wes should se the values here instead https://github.com/calcom/cal.com/blob/main/packages/features/bookings/Booker/components/hooks/useInitialFormValues.ts#L69-L82

It also let's us decide among various sources of name and email where should prevResponse.email's preference lie.

e.g. If a user is loggedIn, shouldn't the loggedIn user's name and email be used instead of using the localstorage one.